src/Form/Extension/PaymentMethodTypeExtension.php

```diff
@@ -4,33 +4,24 @@
 
 namespace CommerceWeavers\SyliusTpayPlugin\Form\Extension;
 
+use CommerceWeavers\SyliusTpayPlugin\Form\EventListener\AddTpayImageFieldsListener;
 use CommerceWeavers\SyliusTpayPlugin\Form\EventListener\SetTpayDefaultPaymentImageUrlListener;
-use CommerceWeavers\SyliusTpayPlugin\Form\Type\PaymentMethodImageType;
 use Sylius\Bundle\PaymentBundle\Form\Type\PaymentMethodType;
 use Symfony\Component\Form\AbstractTypeExtension;
-use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Form\FormEvents;
 
 final class PaymentMethodTypeExtension extends AbstractTypeExtension
 {
-    public function __construct(private readonly SetTpayDefaultPaymentImageUrlListener $setDefaultPaymentImageUrlListener)
-    {
+    public function __construct(
+        private readonly AddTpayImageFieldsListener $addTpayImageFieldsListener,
```

Comment:
With such typing and AddTpayImageFieldsListener being a final class, there is no way to exchange cthe ontent of the class, thus no need for this extraction outside of increasing readability. I would suggest either a lack of final or interface

Reply:
Same probably goes to second listener